src/backend/pipeline/tuplebuf.c

```diff
@@ -614,7 +614,7 @@ TupleBufferTryWait(TupleBufferReader *reader)
 	SpinLockRelease(&buf->mutex);
 
 	if (should_wait)
-		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 0);
+		WaitLatch(reader->proc->latch, WL_LATCH_SET | WL_POSTMASTER_DEATH, 1);
```

Comment:
I guess, it's basically going be like having `list_length(db) * TOTAL_SLOTS* processes in a loop like:
```
while (TupleBufferIsEmpty(buf))
  pg_usleep(1000);
```